sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,692 @@
+     */
+    private HttpDecodedResponse ensureExpectedStatus(final HttpDecodedResponse decodedResponse,
+        final SwaggerMethodParser methodParser, RequestOptions options) {
+        final int responseStatusCode = decodedResponse.getSourceResponse().getStatusCode();
+
+        // If the response was success or configured to not return an error status when the request fails, return the
+        // decoded response.
+        if (methodParser.isExpectedResponseStatusCode(responseStatusCode)
+            || (options != null && options.getErrorOptions().contains(ErrorOptions.NO_THROW))) {
+            return decodedResponse;
+        }
+
+        // Otherwise, the response wasn't successful and the error object needs to be parsed.
+        byte[] responseBytes = decodedResponse.getSourceResponse().getBodyAsByteArray().block();
+        if (responseBytes == null || responseBytes.length == 0) {
+            //  No body, create exception empty content string no exception body object.
+            throw new RuntimeException(instantiateUnexpectedException(
+                methodParser.getUnexpectedException(responseStatusCode), decodedResponse.getSourceResponse(),
+                null, null));
+        } else {
+            Object decodedBody =  decodedResponse.getDecodedBodySync(responseBytes);
+            // create exception with un-decodable content string and without exception body object.
+            throw new RuntimeException(instantiateUnexpectedException(
+                methodParser.getUnexpectedException(responseStatusCode),
+                decodedResponse.getSourceResponse(), responseBytes, decodedBody));
+        }
+    }
+
+    private Object handleRestResponseReturnType(final HttpDecodedResponse response,
+        final SwaggerMethodParser methodParser,
+        final Type entityType) {
+        if (TypeUtil.isTypeOrSubTypeOf(entityType, Response.class)) {
+            final Type bodyType = TypeUtil.getRestResponseBodyType(entityType);
+
+            if (TypeUtil.isTypeOrSubTypeOf(bodyType, Void.class)) {
+                response.getSourceResponse().getBody().ignoreElements().block();
+                return createResponseSync(response, entityType, null);
+            } else {
+                Object bodyAsObject =  handleBodyReturnTypeSync(response, methodParser, bodyType);
+                Response<?> httpResponse = createResponseSync(response, entityType, bodyAsObject);
+                if (httpResponse == null) {
+                    return createResponseSync(response, entityType, null);
+                }
+                return httpResponse;
+            }
+        } else {
+            // For now, we're just throwing if the Maybe didn't emit a value.
+            return handleBodyReturnTypeSync(response, methodParser, entityType);
+        }
+    }
+
+
+    @SuppressWarnings("unchecked")
+    private Response<?> createResponseSync(HttpDecodedResponse response, Type entityType, Object bodyAsObject) {
```

Comment:
Just a nit: do we need to append `Sync` to methods when the class is prepended with `Sync`?